runtime/runtime_test.go

```diff
@@ -2365,6 +2365,62 @@ func TestRuntimeCompositeFunctionInvocationFromImportingProgram(t *testing.T) {
 	require.NoError(t, err)
 }
 
+func TestRuntimeStorageMultipleTransactionsInclusiveRangeFunction(t *testing.T) {
+
+	t.Parallel()
+
+	runtime := newTestInterpreterRuntime()
+
+	inclusiveRangeCreation := []byte(`
+      pub fun createInclusiveRange(): InclusiveRange<Int> {
+        return InclusiveRange(10, 20)
+      }
+    `)
+
+	script1 := []byte(`
+      import "inclusive-range-creation"
+      transaction {
+        prepare(signer: AuthAccount) {
+          let ir = createInclusiveRange()
+          signer.save(ir, to: /storage/inclusiveRange)
+        }
+      }
+    `)
+
+	ledger := newTestLedger(nil, nil)
+
+	runtimeInterface := &testRuntimeInterface{
+		getCode: func(location Location) (bytes []byte, err error) {
+			switch location {
+			case common.StringLocation("inclusive-range-creation"):
+				return inclusiveRangeCreation, nil
+			default:
+				return nil, fmt.Errorf("unknown import location: %s", location)
+			}
+		},
+		storage: ledger,
+		getSigningAccounts: func() ([]Address, error) {
+			return []Address{{42}}, nil
+		},
+	}
+
+	nextTransactionLocation := newTransactionLocationGenerator()
+
+	err := runtime.ExecuteTransaction(
+		Script{
+			Source: script1,
+		},
+		Context{
+			Interface: runtimeInterface,
+			Location:  nextTransactionLocation(),
+		},
+	)
+	RequireError(t, err)
+
+	var subErr *atree.ExternalError
+	require.ErrorAs(t, err, &subErr)
```

Comment:
Assert that the reported error is for the non-storable value:

```suggestion
	var nonStorableValueErr interpreter.NonStorableValueError
	require.ErrorAs(t, err, &nonStorableValueErr)
```

This error is reported quite "late", i.e. at run-time, so might surprise developers.

Maybe consider making also the type non-storable (for now), by changing `InclusiveRangeType.IsStorable()` to return `false`.

Reply:
Updated to make the type non-storable. Now we get a checker `TypeMismatchError`, so I am asserting that - db22c2cc6